### PR TITLE
Psycopg3 Transactions update

### PR DESCRIPTION
**Pull request by nasc17**
Psycopg 3 update, auto commit set to true automatically. Need to check for user defined transaction. 
addressing:  [#433](https://github.com/microsoft/azuredatastudio-postgresql/issues/433)

image

Edit messaging:
image

### CODE REVIEW

File: ossdbtoolsservice/query/batch.py [hunk 1]
```diff
@@ -163,6 +163,10 @@ def save_as(self, params: SaveResultsRequestParams, file_factory: FileStreamFact
 
         self._result_set.save_as(params, file_factory, on_success, on_failure)
 
+    def notice_handler(self, notice: str, conn: ServerConnection):
+        if not (conn.user_transaction and notice.message_primary == 'there is already a transaction in progress'):
```
Comment (DaeunYim) [inline ossdbtoolsservice/query/batch.py:167]: Is there any specific reason why we add the message when the message is 'there is already a transaction in progress'):?

**Review by DaeunYim — APPROVED** [review]
All secnearios worked well for me

File: ossdbtoolsservice/query/batch.py (hunk 1 above)
Comment (nasc17) [reply ossdbtoolsservice/query/batch.py:171]: Added updates to this:
It seems by default Psycopg3 is always starting a transaction. So when executing "begin" command, it sends warning that one has already started. I added a check so we only show this warning when a transaction is being started while still currently in a user started transaction block.
image

**Comment by DaeunYim** [comment]
the new changes lgtm. Is the reason why we have disable_auto_commit and user_transaction because of 
```
if bool(self._execution_plan_options):
                if self._execution_plan_options.include_estimated_execution_plan_xml:
                    sql_statement_text = Query.EXPLAIN_QUERY_TEMPLATE.format(sql_statement_text)
                elif self._execution_plan_options.include_actual_execution_plan_xml:
                    self._disable_auto_commit = True
                    sql_statement_text = Query.ANALYZE_EXPLAIN_QUERY_TEMPLATE.format(sql_statement_text)
```
in query.py? I was wondering why we have two variables for the same purpose.

**Comment by nasc17** [comment]
> the new changes lgtm. Is the reason why we have disable_auto_commit and user_transaction because of
> 
> ```
> if bool(self._execution_plan_options):
>                 if self._execution_plan_options.include_estimated_execution_plan_xml:
>                     sql_statement_text = Query.EXPLAIN_QUERY_TEMPLATE.format(sql_statement_text)
>                 elif self._execution_plan_options.include_actual_execution_plan_xml:
>                     self._disable_auto_commit = True
>                     sql_statement_text = Query.ANALYZE_EXPLAIN_QUERY_TEMPLATE.format(sql_statement_text)
> ```
> 
> in query.py? I was wondering why we have two variables for the same purpose.

We were resetting auto commit based on Analyze Explain, which I did not believe had anything to do with transactions. So added the user transaction property. Also use this particular property to determine if we should force rollback command if error found. (We do not do this for user transaction, raise error to end block manual)

**Pull request merged**